public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx

```diff
@@ -51,35 +49,6 @@ export const useGetGroupOptionsFromFolder = (folderTitle: string) => {
   return { groupOptions, loading: groupfoldersForGrafana?.loading };
 };
 
-const useRuleFolderFilter = (existingRuleForm: RuleForm | null) => {
-  const isSearchHitAvailable = useCallback(
-    (hit: DashboardSearchHit) => {
-      const rbacDisabledFallback = contextSrv.hasEditPermissionInFolders;
-
-      const canCreateRuleInFolder = contextSrv.hasAccessInMetadata(
-        AccessControlAction.AlertingRuleCreate,
-        hit,
-        rbacDisabledFallback
-      );
-
-      const canUpdateInCurrentFolder =
-        existingRuleForm &&
-        hit.folderId === existingRuleForm.id &&
-        contextSrv.hasAccessInMetadata(AccessControlAction.AlertingRuleUpdate, hit, rbacDisabledFallback);
```

Comment:
This is the one bit that might be different after this PR - all the folders that the user has `AlertingRuleCreate` action on will be listed, but we wouldn't list a folder that the user can update rules in but can't create rules in. This would be a really uncommon AC setup though, so I think it's fine not to account for this case. Please correct me if I'm wrong though.

Reply:
I agree this is a relatively uncommon setup. However, if it happens, we'll be displaying totally incorrect values when the user **Edit** an alert rule. We'll show a list of folders with `AlertingRuleCreate` permission, whereas all of them might be wrong (because the user can have edit permission to none of them) 
So, in the described scenario, rule Editing will be completely broken
@yuri-tceretian wdyt?

Reply:
If I'm reading the old code correctly, it would not be completely incorrect values. Previously it would have shown folders where the user can create alerts + the folder that the alert rule that is being updated is in (if the user is allowed to update alerts in it).

After this change, the risk is that we might not show the current folder of the alert rule that is being updated.

It's still a risk though, so if you think that this is bad enough and is likely to be an issue for some users, we'll need to think about another way to solve this.

Reply:
I think the situation when a user can only update rules but not create one is not something uncommon.

I created a role 
<details><summary>role</summary>

```json
{
    "version": 1,
    "uid": "ce004a84-4757-4151-8803-b43469f05c4c",
    "name": "custom:alerting_3",
    "displayName": "Custom Alerting in folder",
    "description": "Control alerts in folder",
    "group": "",
    "permissions": [
        {
            "action": "folders:read",
            "scope": "folders:*",
            "updated": "2023-04-17T15:07:00-04:00",
            "created": "2023-04-17T15:07:00-04:00"
        },
        {
            "action": "datasources:query",
            "scope": "datasources:uid:*",
            "updated": "2023-04-17T15:07:00-04:00",
            "created": "2023-04-17T15:07:00-04:00"
        },
        {
            "action": "alert.rules:write",
            "scope": "folders:uid:d40049df-f4cd-4106-a9ae-fefffd26d8df",
            "updated": "2023-04-17T15:07:00-04:00",
            "created": "2023-04-17T15:07:00-04:00"
        },
        {
            "action": "alert.rules:write",
            "scope": "folders:uid:f750b100-662a-4ccb-9ce8-97bb8e28988d",
            "updated": "2023-04-17T15:07:00-04:00",
            "created": "2023-04-17T15:07:00-04:00"
        },
        {
            "action": "alert.rules:create",
            "scope": "folders:uid:c43e5f3d-be81-4684-b349-a08c8db29bfb",
            "updated": "2023-04-17T15:07:00-04:00",
            "created": "2023-04-17T15:07:00-04:00"
        }
    ],
    "updated": "2023-04-17T15:07:00.722346-04:00",
    "created": "2023-04-17T15:07:00.722346-04:00",
    "global": false
}
```
</details>

that gives write (update) access to two folders and create-only in the 3rd and then assigned to a user with a Viewer base role.

When that user edits a rule in either `f750b1` or `d4004` folders with write access, the dropdown shows only two folders: the current folder that stores the alert and `c43e5` for which user has create-only access. That happens in the main and PR branches.
The only difference I could find is what search API call returns: in PR branch it returns only those 2 folders whereas in main it returns all folders available for read with access control metadata. 

So, I think nothing really changes from alerting editor point of view.

Reply:
I checked again, and yes! The API returned only the one user has create permission but UI displayed both: current rule's folder and the one that user has create access to.

Reply:
Thanks so much for testing this @yuri-tceretian and @gamab! It's great to hear that everything works as it should.